src/stores/workspace/WorkspaceStore.ts

```diff
@@ -157,23 +157,63 @@ export default class WorkspaceStore {
 
 	@action
 	public onSavedItemSelect = (savedItem: EventTreeNode | EventAction | EventMessage) => {
-		if (this.workspacesStore.searchWorkspace === this) {
-			const newWorkspace = this.workspacesStore.createWorkspace({
+		if (this !== this.workspacesStore.searchWorkspace) {
+			if (isEventMessage(savedItem)) {
+				this.viewStore.activePanel = this.messagesStore;
+				this.messagesStore.onMessageSelect(savedItem);
+			} else {
+				this.viewStore.activePanel = this.eventsStore;
+				this.eventsStore.onEventSelect(savedItem);
+			}
+			return;
+		}
+
+		let newWorkspace: WorkspaceStore;
+		new Promise<void>(res => {
```

Comment:
This workaround just as by as my timeout in MessagesStore. Eventually workspace should accept sse filters and properly initialize itself, im working on it now so if this workaround works, its ok for now